File: backend/app/schemas/charts.py
from pydantic import BaseModel


class MarksChartItem(BaseModel):
    subject_id: int
    subject_name: str
    average_marks: float


class MarksChartResponse(BaseModel):
    student_id: int
    data: list[MarksChartItem]


class ExamChartItem(BaseModel):
    exam_id: int
    exam_name: str
    average_marks: float


class ExamChartResponse(BaseModel):
    student_id: int
    data: list[ExamChartItem]


class AttendanceChartResponse(BaseModel):
    student_id: int
    present: int
    absent: int
    total: int
    percentage: float

File: frontend/lib/types/charts.ts
export interface MarksChartResponse {
  student_id: number;
  data: Array<{ subject_id: number; subject_name: string; average_marks: number }>;
}

export interface ExamChartResponse {
  student_id: number;
  data: Array<{ exam_id: number; exam_name: string; average_marks: number }>;
}

export interface AttendanceChartResponse {
  student_id: number;
  present: number;
  absent: number;
  total: number;
  percentage: number;
}
